lib/rules/no-whitespace-before-property.js

```diff
@@ -16,6 +16,42 @@ module.exports = function(context) {
     var sourceCode = context.getSourceCode();
 
     //--------------------------------------------------------------------------
+    // Helpers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Returns the name of the property.
+     * @param {Token} token - The token to check.
+     * @returns {string} The name of the node.
+     * @private
+     */
+    function getPropName(token) {
+        if (token.type === "Literal") {
+            return token.value;
+        } else if (token.type === "Identifier") {
+            return token.name;
+        }
+        return false;
```

Comment:
`false` is not a string (jsdoc says return type is string).
IMO, we can use `sourceCode.getText(prop)` or a constant text such as `"computed property"` instead of this function.

Could you add tests?: `foo["on" + name]`